app/repositories/scores.py

```diff
@@ -1,47 +1,90 @@
 from __future__ import annotations
 
-import textwrap
 from datetime import datetime
-from typing import Any
 from typing import TypedDict
 from typing import cast
 
+from sqlalchemy import Column
+from sqlalchemy import DateTime
+from sqlalchemy import Index
+from sqlalchemy import Integer
+from sqlalchemy import String
+from sqlalchemy import func
+from sqlalchemy import insert
+from sqlalchemy import select
+from sqlalchemy import update
+from sqlalchemy.dialects.mysql import FLOAT
+from sqlalchemy.dialects.mysql import TINYINT
+
 import app.state.services
 from app._typing import UNSET
 from app._typing import _UnsetSentinel
+from app.repositories import DIALECT
+from app.repositories import Base
+
+
+class ScoresTable(Base):
+    __tablename__ = "scores"
+
+    id = Column("id", Integer, primary_key=True)
+    map_md5 = Column("map_md5", String(32), nullable=False)
+    score = Column("score", Integer, nullable=False)
+    pp = Column("pp", FLOAT(precision=6, scale=3), nullable=False)
+    acc = Column("acc", FLOAT(precision=6, scale=3), nullable=False)
+    max_combo = Column("max_combo", Integer, nullable=False)
+    mods = Column("mods", Integer, nullable=False)
+    n300 = Column("n300", Integer, nullable=False)
+    n100 = Column("n100", Integer, nullable=False)
+    n50 = Column("n50", Integer, nullable=False)
+    nmiss = Column("nmiss", Integer, nullable=False)
+    ngeki = Column("ngeki", Integer, nullable=False)
+    nkatu = Column("nkatu", Integer, nullable=False)
+    grade = Column("grade", String(2), nullable=False, default="N")
+    status = Column("status", Integer, nullable=False)
+    mode = Column("mode", Integer, nullable=False)
+    play_time = Column("play_time", DateTime, nullable=False)
+    time_elapsed = Column("time_elapsed", Integer, nullable=False)
+    client_flags = Column("client_flags", Integer, nullable=False)
+    userid = Column("userid", Integer, nullable=False)
+    perfect = Column("perfect", TINYINT(1), nullable=False)
+    online_checksum = Column("online_checksum", String(32), nullable=False)
+
+    __table_args__ = (
+        Index("scores_map_md5_index", map_md5),
+        Index("scores_score_index", score),
+        Index("scores_pp_index", pp),
+        Index("scores_mods_index", mods),
+        Index("scores_status_index", status),
+        Index("scores_mode_index", mode),
+        Index("scores_play_time_index", play_time),
+        Index("scores_userid_index", userid),
+        Index("scores_online_checksum_index", online_checksum),
```

Comment:
ouu didn't notice, this should be a unique key at the ddl level

can make the change in another pr